client/client.go

```diff
@@ -303,8 +303,9 @@ func (r *NotaryRepository) RemoveTarget(targetName string) error {
 	return nil
 }
 
-// ListTargets lists all targets for the current repository
-func (r *NotaryRepository) ListTargets() ([]*Target, error) {
```

Comment:
Have another look. Both the prioritization and returned data should make more sense now.